ctru-rs/src/services/am.rs

```diff
@@ -0,0 +1,121 @@
+use crate::error::ResultCode;
+use crate::services::fs::FsMediaType;
+use std::marker::PhantomData;
+use std::mem::MaybeUninit;
+
+#[derive(Copy, Clone, Debug)]
+#[repr(C)]
+pub struct TitleInfo {
+    id: u64,
```

Comment:
This struct shouldn't be a copy of already existing (and auto-updating) types. You should simply hold a `ctru_sys::AM_TitleEntry` in here and reference that instead.

Reply:
Done